src/main/java/io/supertokens/storage/mysql/Start.java

```diff
@@ -1192,16 +1212,23 @@ public void updateUserEmail_Transaction(TransactionConnection con, String userId
             throws StorageQueryException, UnknownUserIdException, DuplicateEmailException {
         Connection sqlCon = (Connection) con.getConnection();
         try {
-            PasswordlessQueries.updateUserEmail_Transaction(this, sqlCon, userId, email);
-        } catch (SQLException e) {
-            // TODO: review these exceptions
+            int updated_rows = PasswordlessQueries.updateUserEmail_Transaction(this, sqlCon, userId, email);
+            if (updated_rows != 1) {
+                throw new UnknownUserIdException();
+            }
+        } catch (UnknownUserIdException | SQLException e) {
+
+            if (e instanceof UnknownUserIdException) {
+                throw new UnknownUserIdException();
+            }
```

Comment:
You don't need to catch `UnknownUserIdException` and then throw the exact same thing again. Just don't catch `UnknownUserIdException` :)

Reply:
Agreed, I missed that.